tfhe/src/core_crypto/commons/generators/encryption/mod.rs

```diff
@@ -524,86 +526,21 @@ impl<G: ParallelByteRandomGenerator> EncryptionRandomGenerator<G> {
         let mask_iter = self
             .mask
             .par_fork_lwe_compact_ciphertext_list_to_bin::<T>(lwe_mask_count, lwe_dimension)?;
-        let noise_bytes = noise_bytes_per_lwe_compact_ciphertext_bin(lwe_dimension);
-        self.par_try_fork(lwe_mask_count.0, mask_iter, noise_bytes)
+        let noise_iter = self
+            .noise
+            .par_fork_lwe_compact_ciphertext_list_to_bin(lwe_mask_count, lwe_dimension)?;
+        self.par_map_to_encryption_generator(mask_iter, noise_iter)
     }
 
     // Forks both generators into a parallel iterator.
-    fn par_try_fork(
+    fn par_map_to_encryption_generator(
         &mut self,
-        n_child: usize,
         mask_iter: impl IndexedParallelIterator<Item = MaskRandomGenerator<G>>,
-        noise_bytes: usize,
+        noise_iter: impl IndexedParallelIterator<Item = NoiseRandomGenerator<G>>,
     ) -> Result<impl IndexedParallelIterator<Item = EncryptionRandomGenerator<G>>, ForkError> {
```

Comment:
Result could be removed from signature
Not sure if this is useful as it forces callers to wrap in an `Ok` themselves

Reply:
Disagree, when wrong number of bytes/children are computed we can and will fail the fork, it's a very useful tool when debugging